packages/gatsby/src/utils/webpack.config.js

```diff
@@ -56,7 +56,20 @@ module.exports = async (
   // We combine develop & develop-html stages for purposes of generating the
   // webpack config.
   const stage = suppliedStage
-  const { rules, loaders, plugins } = createWebpackUtils(stage, program)
+
+  const isRspack = stage === `build-javascript` || stage === `develop`
+
+  const { rules, loaders, plugins } = createWebpackUtils(stage, program, {
+    isRspack,
+  })
+
+  let rsPackDefine = {}
+  if (isRspack) {
+    plugins.define = someObject => {
+      rsPackDefine = { ...rsPackDefine, ...someObject }
+      return undefined
+    }
```

Comment:
Using `builtins.define` instead of `DefinePlugin` ( https://www.rspack.dev/config/builtins.html#builtinsdefine / https://www.rspack.dev/guide/plugin-compat.html )